### DISCUSSION TITLE
Drop-down menu

### DISCUSSION THREAD

**Discussion by marcelo-moglie** [bonsai-rx/bonsai#1248] [Q&A]
Hi, is there any way to create a drop-down menu for an externalized property such that the values are loaded beforehand and then displayed in the menu for the user to select them?

Thanks!

**Comment by bruno-f-cruz** [comment]
Heya, 
Currently, the easiest option is to define an `enum` in a scripting c# script and expose it as a property of the node.
E.g.

```csharp
using Bonsai;
using System;
using System.ComponentModel;
using System.Collections.Generic;
using System.Linq;
using System.Reactive.Linq;

[Combinator]
[Description("")]
[WorkflowElementCategory(ElementCategory.Source)]
public class MyDropdown
{
    public DropdownOptions OptionSelected { get; set; }
    public IObservable<int> Process()
    {
        return Observable.Return(0);
    }
}


public enum DropdownOptions{
    Option1,
    Option2,
    Option3,
}
```

Hope it helps!

**Reply by marcelo-moglie** [reply]
Hi Bruno,

Thanks a lot for your reply. I've been able to generate the script using the CSharpSource node but I haven't been able to expose it as a property of a node afterwords. Could you please guide me on how to do it?

Many thanks,
Marcelo

**Reply by bruno-f-cruz** [reply]
Did you copy-paste the example I sent you? In my Bonsai workflow, `DropdownOptions` appear under a property of my node.

**Reply by marcelo-moglie** [reply]
Now it is working!! I was not naming correctly the files and nodes. Thanks a lot!!
